jdbc/src/main/clojure/xtdb/next/jdbc.clj

```diff
@@ -20,11 +20,22 @@
                    (String. StandardCharsets/UTF_8)))))
 
 (defn- denormalize-keys [v, ^IKeyFn key-fn]
-  (w/prewalk (fn [v]
-               (cond-> v
-                 (instance? Map v)
-                 (update-keys #(.denormalize key-fn %))))
-             v))
+  (letfn [(process-map [m]
+            (if (keyword? (first (keys m)))
```

Comment:
what's the reasoning (/assumption) behind only checking one key of the map here?

Reply:
resolved by delegating to prewalk which in practice confirmed your theory :)